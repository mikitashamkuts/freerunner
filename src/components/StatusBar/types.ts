import {CustomComponentProps} from '../../types';

export interface Props extends CustomComponentProps {
  style: 'light' | 'dark';
}
