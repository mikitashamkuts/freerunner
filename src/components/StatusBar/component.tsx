import React, {FC, memo} from 'react';
import {StatusBar as RNStatusBar} from 'react-native';

import {whyDidItRenderConfig} from '../../../debug';

import {Props} from '.';

const StatusBar: FC<Props> = ({style = 'light'}) => {
  return (
    <RNStatusBar
      translucent={true}
      animated={true}
      barStyle={`${style}-content`}
      backgroundColor="transparent"
    />
  );
};

StatusBar.whyDidYouRender = whyDidItRenderConfig.UIComponentDebugActive;

export default memo(StatusBar);
