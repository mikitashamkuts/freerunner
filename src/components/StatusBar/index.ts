export {default as StatusBar} from './component';
export * from './types';
