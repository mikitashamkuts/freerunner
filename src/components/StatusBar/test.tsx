/**
 * @format
 */

import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

import StatusBar from './component';

describe('StatusBar Component', () => {
  it('Renders with default props and matches previous snapshot: ', () => {
    const tree = renderer.create(<StatusBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders with light style and matches previous snapshot: ', () => {
    const tree = renderer.create(<StatusBar customStyle="light" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders with dark style and matches previous snapshot: ', () => {
    const tree = renderer.create(<StatusBar customStyle="dark" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
