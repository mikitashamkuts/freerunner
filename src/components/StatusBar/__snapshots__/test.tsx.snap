// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar Component Renders with dark style and matches previous snapshot:  1`] = `null`;

exports[`StatusBar Component Renders with default props and matches previous snapshot:  1`] = `null`;

exports[`StatusBar Component Renders with light style and matches previous snapshot:  1`] = `null`;
